lib/commands/condition.js

```diff
@@ -0,0 +1,128 @@
+ *             "description": "Name: 100% Loss Scenario\n
+ *                             Downlink Bandwidth: 0 Mbps\n
+ *                             Downlink Latency:0 ms\n
+ *                             Downlink Packet Loss Ratio: 100%\n
+ *                             Uplink Bandwidth: 0 Mbps\n
+ *                             Uplink Latency: 0 ms\n
+ *                             Uplink Packet Loss Ratio: 100%"
+ *         }
+ *     ],
+ *     "profilesSorted": true,
+ *     "identifier": "SlowNetworkCondition",    // MobileEnableConditionInducer conditionID
+ *     "isDestructive": false,
+ *     "isInternal": false,
+ *     "activeProfile": "",
+ *     "name": "Network Link",
+ *     "isActive": false
+ * }]
+ */
+commands.mobileListConditionInducerProfiles = async function listConditionInducerProfiles () {
+  requireConditionInducerCompatibleDevice(this);
+  const conditionInducerService = await services.startInstrumentService(this.opts.device.udid);
+  try {
+    const ret = await conditionInducerService.callChannel(INSTRUMENT_CHANNEL.CONDITION_INDUCER, 'availableConditionInducers');
+    return ret.selector;
+  } finally {
+    conditionInducerService.close();
+  }
+};
+
+/**
+ * Enable condition inducer. The socket needs to remain connected during operation
+ * conditionID  ProfileList[0].identifier
+ * profileID    ProfileList[0].profiles[0].identifier
+ * @param {{conditionID,profileID}} options
+ * @returns {boolean} `true` if enable the condition succeeded
+ * @throws {Error} If try to start another Condition and the previous Condition has not stopped, it will throw an error
+ */
+commands.mobileEnableConditionInducer = async function enableConditionInducer (options = {}) {
+  requireConditionInducerCompatibleDevice(this);
+  if (this._conditionInducerService && !this._conditionInducerService._socketClient.destroyed) {
+    this.log.errorAndThrow(`Condition inducer has been started. A condition is already active.`);
+  }
+  const {
+    conditionID,
+    profileID,
+  } = options;
+  this._conditionInducerService = await services.startInstrumentService(this.opts.device.udid);
+  const ret = await this._conditionInducerService.callChannel(INSTRUMENT_CHANNEL.CONDITION_INDUCER,
+    'enableConditionWithIdentifier:profileIdentifier:', conditionID, profileID);
+  if (!_.isBoolean(ret.selector)) {
+    this._conditionInducerService.close();
+    this._conditionInducerService = null;
+    this.log.errorAndThrow(`Enable condition inducer error: '${JSON.stringify(ret.selector)}'`);
+  }
+  return ret.selector;
+};
+
+/**
+ * Disable condition inducer
+ * this call upon driver deletion
+ * @returns {boolean} `true` if disable the condition succeeded
+ */
+commands.mobileDisableConditionInducer = async function disableConditionInducer () {
+  try {
+    const ret = await this._conditionInducerService.callChannel(INSTRUMENT_CHANNEL.CONDITION_INDUCER, 'disableActiveCondition');
```

Comment:
shall we check first if this._conditionInducerService is defined? exceptions-controlled flows are usually not the best chice